tests/FSharp.Compiler.ComponentTests/Conformance/UnionTypes/MultiCaseUnionStructTypes.fs

```diff
@@ -0,0 +1,250 @@
+        Fsx """
+namespace Foo
+[<Struct>]
+type StructUnion =
+    | A of int * string
+    | B of b: string
+    | C of string
+        """
+        |> compile
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 3204, Line 5, Col 12, Line 5, Col 15, "If a union type has more than one case and is a struct, then all fields within the union type must be given unique field names. For example: 'type A = B of b: int | C of c: int' (unique field names 'b' and 'c' assigned).")
+            (Error 3204, Line 5, Col 18, Line 5, Col 24, "If a union type has more than one case and is a struct, then all fields within the union type must be given unique field names. For example: 'type A = B of b: int | C of c: int' (unique field names 'b' and 'c' assigned).")
+            (Error 3204, Line 7, Col 12, Line 7, Col 18, "If a union type has more than one case and is a struct, then all fields within the union type must be given unique field names. For example: 'type A = B of b: int | C of c: int' (unique field names 'b' and 'c' assigned).")
+        ]
+        
+    [<Fact>]
+    let ``If a union type has more than one case and is a struct, field must be given unique name 13`` () =
+        Fsx """
+namespace Foo
+[<Struct>]
+type StructUnion =
+    | A of a: int * string
+    | B of string * b: string
+    | C of c: string * string * c3: int
+        """
+        |> compile
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 3204, Line 5, Col 21, Line 5, Col 27, "If a union type has more than one case and is a struct, then all fields within the union type must be given unique field names. For example: 'type A = B of b: int | C of c: int' (unique field names 'b' and 'c' assigned).")
+            (Error 3204, Line 6, Col 12, Line 6, Col 18, "If a union type has more than one case and is a struct, then all fields within the union type must be given unique field names. For example: 'type A = B of b: int | C of c: int' (unique field names 'b' and 'c' assigned).")
+            (Error 3204, Line 7, Col 24, Line 7, Col 30, "If a union type has more than one case and is a struct, then all fields within the union type must be given unique field names. For example: 'type A = B of b: int | C of c: int' (unique field names 'b' and 'c' assigned).")
+        ]
+        
+    [<Fact>]
+    let ``If a union type has more than one case and is a struct, field must be given unique name. involves an immediate cyclic reference`` () =
+        Fsx """
+namespace Foo
+[<Struct>]
+type StructUnion = A of X:int | B of Y:StructUnion
+        """
+        |> compile
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 954, Line 4, Col 6, Line 4, Col 17, "This type definition involves an immediate cyclic reference through a struct field or inheritance relation")
+        ]
+        
+    [<Fact>]
+    let ``If a union type has more than one case and is a struct, field must be given unique name 14`` () =
+        Fsx """
```

Comment:
This thing works as of now (since there is Item1 generated, but it does not conflict).
We should keep it working (backwards compat).

https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AbEAzAzmgExAGoAfAOwEMBbGXAB0rBgAIAxCCAWACgBtADwBlAC5QArmBEA+ALq8RAT3qsAchBEBBACqxWAXl4tjLUiwCi1ekqMmzAYQAWASwwFY5FhGwtcYySIsABTO5IEAVL5ioQDmAJRAA=

Reply:
Updated .